crates/nu-color-config/src/style_computer.rs

```diff
@@ -0,0 +1,285 @@
+            _ => Style::default(),
+        }
+    }
+
+    // Used only by the `table` command.
+    pub fn style_primitive(&self, value: &Value) -> TextStyle {
+        let s = self.compute(&value.get_type().to_string(), value);
+        match *value {
+            Value::Bool { .. } => TextStyle::with_style(AlignmentHorizontal::Left, s),
+
+            Value::Int { .. } => TextStyle::with_style(AlignmentHorizontal::Right, s),
+
+            Value::Filesize { .. } => TextStyle::with_style(AlignmentHorizontal::Right, s),
+
+            Value::Duration { .. } => TextStyle::with_style(AlignmentHorizontal::Left, s),
+
+            Value::Date { .. } => TextStyle::with_style(AlignmentHorizontal::Left, s),
+
+            Value::Range { .. } => TextStyle::with_style(AlignmentHorizontal::Left, s),
+
+            Value::Float { .. } => TextStyle::with_style(AlignmentHorizontal::Right, s),
+
+            Value::String { .. } => TextStyle::with_style(AlignmentHorizontal::Left, s),
+
+            Value::Nothing { .. } => TextStyle::with_style(AlignmentHorizontal::Left, s),
+
+            Value::Binary { .. } => TextStyle::with_style(AlignmentHorizontal::Left, s),
+
+            Value::CellPath { .. } => TextStyle::with_style(AlignmentHorizontal::Left, s),
+
+            Value::Record { .. } | Value::List { .. } | Value::Block { .. } => {
+                TextStyle::with_style(AlignmentHorizontal::Left, s)
+            }
+            _ => TextStyle::basic_left(),
+        }
+    }
+
+    // The main constructor.
+    pub fn from_config(engine_state: &'a EngineState, stack: &'a Stack) -> StyleComputer<'a> {
+        let config = engine_state.get_config();
+
+        // Create the hashmap
+        let mut map: StyleMapping = HashMap::from([
+            initial!("separator", Color::White.normal()),
+            initial!(
+                "leading_trailing_space_bg",
+                Style::default().on(Color::Rgb(128, 128, 128))
+            ),
+            initial!("header", Color::White.normal()),
+            initial!("empty", Color::White.normal()),
+            initial!("bool", Color::White.normal()),
+            initial!("int", Color::White.normal()),
+            initial!("filesize", Color::White.normal()),
+            initial!("duration", Color::White.normal()),
+            initial!("date", Color::White.normal()),
+            initial!("range", Color::White.normal()),
+            initial!("float", Color::White.normal()),
+            initial!("string", Color::White.normal()),
+            initial!("nothing", Color::White.normal()),
+            initial!("binary", Color::White.normal()),
+            initial!("cellpath", Color::White.normal()),
+            initial!("row_index", Color::Green.bold()),
+            initial!("record", Color::White.normal()),
+            initial!("list", Color::White.normal()),
+            initial!("block", Color::White.normal()),
+            initial!("hints", Color::DarkGray.normal()),
```

Comment:
Seeing these all here made me wonder if this list was complete. So, I checked here https://github.com/nushell/nushell/blob/2bacc29d30fc6bb4798c9ef99801267492feb80d/crates/nu-parser/src/parser.rs#L2723-L2747, which may not be quite up to date either, considering this https://github.com/nushell/nushell/blob/2bacc29d30fc6bb4798c9ef99801267492feb80d/crates/nu-protocol/src/syntax_shape.rs#L9-L108.

At some point we need to get all these in sync, but we can save that for another PR.